services/madoc-ts/src/routes/projects/update-revision-task.ts

```diff
@@ -0,0 +1,36 @@
+import { api } from '../../gateway/api.server';
+import { userWithScope } from '../../utility/user-with-scope';
+import { RouteMiddleware } from '../../types/route-middleware';
+import { CrowdsourcingTask } from '../../gateway/tasks/crowdsourcing-task';
+import { userCan } from '../../utility/user-can';
+
+export const updateRevisionTask: RouteMiddleware<{ taskId: string; task: any }> = async context => {
+  const { siteId } = userWithScope(context, ['models.contribute', 'models.revision']);
+
+  const canCreate = userCan('models.create', context.state);
+  const canReview = userCan('models.revision', context.state);
+
+  const userApi = api.asUser({ siteId });
+  const id = context.params.taskId;
+  const taskBody = context.requestBody.task;
+
+  const task = await userApi.getTask(id);
```

Comment:
One last thing, the user _might_ be able to edit other users contributions. I think the task has an `assignee` property that we could check if they `canCreate` but can't review. 

There is also `delegated_owners` which _should_ contain the assigned reviewer. (But I can't remember if that's 100% working, it should allow reviewers who are assigned to the review task to progress the "delegated" task, which is crowdsourcing task)

Reply:
I think thats checked in [update-revision.ts ](https://github.com/digirati-co-uk/madoc-platform/blob/NS-17/hotfix/review-issues/services/madoc-ts/src/capture-model-server/routes/update-revision.ts) ?